app/controllers/candidate_interface/personal_details_controller.rb

```diff
@@ -21,12 +22,30 @@ def update
     end
 
     def show
-      personal_details_form = PersonalDetailsForm.build_from_application(
-        current_application,
-      )
+      @application_form = current_application
+      personal_details_form = PersonalDetailsForm.build_from_application(current_application)
       @personal_details_review = PersonalDetailsReviewPresenter.new(form: personal_details_form)
     end
 
+    def complete
+      @application_form = current_application
+
+      if PersonalDetailsForm.build_from_application(current_application).valid?
+        current_application.update!(application_form_params)
+
+        redirect_to candidate_interface_application_form_path
+      else
```

Comment:
I don't think this `else` clause can trigger under normal use. The user can't reach this form if they haven't filled in their personal details successfully. Do we need it?

Reply:
Yea I meant to left a comment saying the else should never trigger. It can come out 💯

Reply:
New version is shorter and nicer! 👍